rados/ioctx.go

```diff
@@ -122,6 +149,45 @@ func (ioctx *IOContext) Append(oid string, data []byte) error {
 	return GetRadosError(int(ret))
 }
 
+// Executes read operation with librados operation flags
+func (ioctx *IOContext) ReadOperate(oid string, data []byte,
+	offset uint64) (int, error) {
+	if len(data) == 0 {
+		return 0, nil
+	}
+
+	c_oid := C.CString(oid)
+	defer C.free(unsafe.Pointer(c_oid))
+
+	r_op := C.rados_create_read_op()
+	if unsafe.Pointer(r_op) == nil {
+		return 0, GetRadosError(C.EIO)
+	}
+	defer C.rados_release_read_op(r_op)
+
+	var bytesRead uint64
+	var preval int
+	C.rados_read_op_read(r_op,
+		(C.uint64_t)(offset),
+		(C.size_t)(len(data)),
+		(*C.char)(unsafe.Pointer(&data[0])),
+		(*C.size_t)(unsafe.Pointer(&bytesRead)),
+		(*C.int)(unsafe.Pointer((&preval))))
+
+	if preval < 0 {
+		return 0, GetRadosError(preval)
+	}
+
+	ret := C.rados_read_op_operate(r_op, ioctx.ioctx, c_oid,
+		(C.int)(LIBRADOS_OP_FULL_TRY))
```

Comment:
Add a comment what exactly this flag does.

Reply:
Done.